assets/js/components/user-input/UserInputQuestionnaire.js

```diff
@@ -73,8 +73,28 @@ export default function UserInputQuestionnaire() {
 	);
 
 	useEffect( () => {
-		if ( activeSlug === 'preview' ) {
-			trackEvent( viewContext, 'summary_view' );
+		switch ( activeSlug ) {
+			case USER_INPUT_QUESTIONS_PURPOSE:
+				trackEvent(
+					`${ viewContext }_kmw`,
+					'site_purpose_question_view'
+				);
+				break;
+			case USER_INPUT_QUESTION_POST_FREQUENCY:
+				trackEvent(
+					`${ viewContext }_kmw`,
+					'content_frequency_question_view'
+				);
+				break;
+			case USER_INPUT_QUESTIONS_GOALS:
+				trackEvent(
+					`${ viewContext }_kmw`,
+					'site_goals_question_view'
+				);
+				break;
+			case 'preview':
+				trackEvent( `${ viewContext }_kmw`, 'summary_view' );
+				break;
 		}
```

Comment:
Not sure if this is actually more concise, but it's a bit less nesting and repeats the `trackEvent( `${ viewContext }_kmw`, ` repeating itself a bunch:

```suggestion
		// Set the event name to track based on the active slug.
		let eventActionName;
		if ( activeSlug === USER_INPUT_QUESTIONS_PURPOSE) {
			eventActionName = 'site_purpose_question_view';
		}
		if ( activeSlug === USER_INPUT_QUESTION_POST_FREQUENCY) {
			eventActionName = 'content_frequency_question_view';
		}
		if ( activeSlug === USER_INPUT_QUESTIONS_GOALS) {
			eventActionName = 'site_goals_question_view';
		}
		if ( activeSlug === 'preview') {
			eventActionName = 'summary_view';
		}

		if ( eventActionName ) {
			trackEvent( `${ viewContext }_kmw`, eventActionName );
		}
```

What do you think?